backend/src/definitions/pipeline.ts

```diff
@@ -1,4 +1,50 @@
 import { z } from "zod";
+import { ExternalColumnSchema } from ".";
+
+// SCHEMA INFERENCE:
+export const InferredSchemaColumnSchema = ExternalColumnSchema.extend({
+  table: z.string(),
+});
+export type InferredSchemaColumn = z.infer<typeof InferredSchemaColumnSchema>;
+
+// A relation used in a pipeline. Includes the API path to the relation from the base object defintiion and the object definition ID of the related object defintion
+export const InferredSchemaRelationSchema = z.object({
+  api_path: z.string(),
```

Comment:
Is API path still relevant?

Reply:
It is not. Additional changes to these schemas and types (including removing unneeded ones) will be made as part of the query builder PR.